src/main/cpp/states/Music.cpp

```diff
@@ -0,0 +1,107 @@
+#include <states/Music.h>
+
+int Music::GetButton() 
+{
+    for (int i = 1; i < 9; i++) 
+    {
+        if (m_joystick->GetRawButton(i))
+        {
+            return i;
+        }
+    }
+    return 0;
+}
+
+void Music::LoadMusicSelection(int offset)
+{
+    songSelection += offset;
+
+    if (songSelection >= SONG_COUNT)
+    {
+        songSelection = 0;
+    }
+
+    if (songSelection < 0)
+    {
+        songSelection = SONG_COUNT - 1;
+    }
+
+    m_orchestra -> LoadMusic(songs[songSelection]);
+    timeToPlayLoops = 10;
+}
+
+void Music::Init()
+{
+    m_orchestra = new Orchestra();
+
+    m_talon = new TalonFX * [TALON_COUNT];

```

Comment:
Don't construct a falconfx here rather take one in or maybe better yet have the play call also take the falconfx.